src/AutoRest.CSharp/Common/Output/Expressions/Statements/ForeachStatement.cs

```diff
@@ -9,25 +9,30 @@
 
 namespace AutoRest.CSharp.Common.Output.Expressions.Statements
 {
-    internal record ForeachStatement(CodeWriterDeclaration Item, ValueExpression Enumerable, bool IsAsync = false) : MethodBodyStatement, IEnumerable<MethodBodyStatement>
+    internal record ForeachStatement(CSharpType? ItemType, CodeWriterDeclaration Item, ValueExpression Enumerable, bool IsAsync = false, bool UseVarAsItemType = false) : MethodBodyStatement, IEnumerable<MethodBodyStatement>
     {
         private readonly List<MethodBodyStatement> _body = new();
         public IReadOnlyList<MethodBodyStatement> Body => _body;
 
-        public ForeachStatement(string itemName, EnumerableExpression enumerable, out ValueExpression item)
-            : this(new CodeWriterDeclaration(itemName), enumerable)
+        public ForeachStatement(CSharpType itemType, string itemName, ValueExpression enumerable, out VariableReference item, bool isAsync = false, bool useVarAsItemType = false) : this(itemType, new CodeWriterDeclaration(itemName), enumerable, isAsync, useVarAsItemType)
```

Comment:
Please move `out` parameter to the end of the signature - I think we can make `isAsync` required here.

Reply:
Done